spec/currency_spec.rb

```diff
@@ -37,6 +37,65 @@
     it "raises when the currency is nil" do
       expect { Money::Currency.new(nil) }.to raise_error(Money::Currency::UnknownCurrency)
     end
+    
+    it "looks up crypto currencies when enabled" do
+      config = Money::Config.new.tap do |config|
+        config.experimental_crypto_currencies = true
+      end
+      
+      allow(Money::Currency).to receive(:currencies).and_return({})
+      crypto_currencies = { 
+        "usdc" => { 
+          "iso_code" => "USDC", 
+          "name" => "USD Coin",
+          "symbol" => "USDC",
+          "disambiguate_symbol" => "USDC",
+          "subunit_to_unit" => 100,
+          "smallest_denomination" => 1,
+          "decimal_mark" => "."
+        } 
+      }
+      allow(Money::Currency).to receive(:crypto_currencies).and_return(crypto_currencies)
+      
+      with_config(config) do
+        currency = Money::Currency.new('USDC')
+        expect(currency.iso_code).to eq('USDC')
+        expect(currency.symbol).to eq('USDC')
+      end
+    end
+    
+    it "doesn't look up crypto currencies when disabled" do
+      # Reset class variables to ensure clean state
+      old_currencies = Money::Currency.class_variable_get(:@@loaded_currencies) rescue {}
+      Money::Currency.class_variable_set(:@@loaded_currencies, {})
```

Comment:
question: Is it worth adding this (and the clean-up at the end of the test) to `with_config`?

Reply:
I agree, this test should be improved, it looks like it was generated by AI. The test can be as simple as
```
configure(experimental_crypto_currencies: false) do 
  assert_nil Money::Currency.find("USDC") 
end
```

Reply:
@elfassy - This test wasn't generated by AI, the need for doing the above is because `Money::Currency` has the class variable `@@loaded_currencies` to cache the currencies therefore we need to clear. But yes, we moved the above into the configure method and the test has now become the above.